Added new theme Aurora and theme name duplication validation script.

TriDiamond opened the pull request:
- I have a theme called Obsidian, since I am adding a new theme, the original website will host the new theme. Therefore Obsidian's website will change, so I have edited that as well.

## Check List

**Please read and check followings before submitting a PR.**

- [x] I want to publish my theme on Hexo official website.
    - [x] I have read the [theme publishing doc](https://hexo.io/docs/themes#Publishing).
    - [x] `name` is unique.
    - [x] `link` URL is correct.
    - [x] `preview` URL is correct.
    - [x] `preview` URL web site is rendered correctly.
    - [x] Add a screenshot to `source/themes/screenshots`.
    - [x] Screenshot filename is same as value of `name`.
    - [x] Screenshot size is `800 * 500`.
    - [x] Screenshot file format is `png`.
- [ ] I want to publish my plugin on Hexo official website.
    - [ ] I have read the [plugin publishing doc](https://hexo.io/docs/plugins#Publishing).
    - [ ] `name` is unique.
    - [ ] `link` URL is correct.
- [ ] Others (Update, fix, translation, etc...)
    - Languages:
    - [ ] `en` English
    - [ ] `ko` Korean
    - [ ] `pt-br` Brazilian Portuguese
    - [ ] `ru` Russian
    - [ ] `th` Thai
    - [ ] `zh-cn` simplified Chinese
    - [ ] `zh-tw` traditional Chinese

<!-- 
    Thank you for publishing your work on Hexo site!
    
    If you also would like to become a Hexojs org memeber, here is the opportunity. Simply transfer your repo into Hexojs org, and you will become hexojs member. You could still be the repo admin, but also gain access to hexojs other repoes. 
    
    There are several benefits to do so:
    1. Become Hexojs org member, and gain access to all hexojs repos.
    2. Other Hexojs members could help to maintain issues and review PRs.
    3. More wait you to discover... :)
    
    Please contact hi@abnerchou.me if you are interested in this opportunity.
-->

stevenjoezhang commented:
The screenshot of hexo-theme-pure `source/themes/screenshots/pure.png` is removed in this pull request

TriDiamond commented:
@stevenjoezhang I will try to fork the repo to see if this problem still exist. This file name conflict is due to the capital letter check due to different operation system.

stevenjoezhang commented:
Yes, I reproduced this capital letter issue on macOS

Maybe you can revert the first commit (https://github.com/hexojs/site/pull/1630/commits/27ed9afd64f7fdd66ca971b04d92b0875cb19309) and then re-execute `git add` and `git commit`?

TriDiamond commented:
> Yes, I reproduced this capital letter issue on macOS
> 
> Maybe you can revert the first commit ([27ed9af](https://github.com/hexojs/site/commit/27ed9afd64f7fdd66ca971b04d92b0875cb19309)) and then re-execute `git add` and `git commit`?

@stevenjoezhang It's done, see if it's ok to merge, thanks!

stevenjoezhang commented on scripts/validate.js:
```diff
@@ -1,38 +1,67 @@
+async function validateThemeNames() {
+  let isValidationPassed = true
+  const themeDataFile = fs.readFileSync(hexo.source_dir + '_data/themes.yml')
+  const themeData = yaml.load(themeDataFile)
```
Use `const themeData = hexo.locals.get('data').themes`

TriDiamond replied:
Changes are done, thanks for the suggestion.

TriDiamond commented:
What I did here in my latest commit is:
- First added a validation script on npm build
  - Read all theme from _data/theme.yml
  - Force all theme name to lower cases and check for duplication
- Changed the original theme `pure` to `Pure` to avoid further image conflicting
  - Re-added the original `pure.png` back to the screenshots folder, changed the name to `Pure.png` (Image was found in the original pull request.)
- Changed the later theme `Pure` to `A-Pure` to avoid conflicting with the older theme `pure`
  - Changed the screenshot image name to `A-Pure.png`
  
 > Use `const themeData = hexo.locals.get('data').themes`
 
 Ok will go change that. Thanks for suggesting, didn't know there is a Hexo function for this.

stevenjoezhang commented:
@TriDiamond Thanks very much! We have some discussions about the theme name issue: https://github.com/hexojs/site/issues/837 https://github.com/hexojs/site/pull/1458
I have opened another pull request to use GitHub actions to validate https://github.com/hexojs/site/pull/1647
This will remind new theme publishers whether the theme name and screenshots they submit meet the requirements

Besides, all the semicolons are removed in `scripts/validate.js` and may fail the eslint test

TriDiamond commented:
> @TriDiamond Thanks very much! We have some discussions about the theme name issue: #837 #1458
> I have opened another pull request to use GitHub actions to validate #1647
> This will remind new theme publishers whether the theme name and screenshots they submit meet the requirements
> 
> Besides, all the semicolons are removed in `scripts/validate.js` and may fail the eslint test

Ah yes, I have my prettier setup in my VSCode to remove all semicolons. I will add those back for now. This script can be removed after the GitHub actions is integrated, that approach is definitely better.

TriDiamond commented:
@stevenjoezhang I have fixed all the eslint issue. I ran the eslint script everything is going ok now. Hope this pull request can go through. Once your github actions's pull request had gone through, feel free to remove the theme name check script :) Or do you want me to remove it from this pull request instead? I am happy to do so too.

stevenjoezhang commented:
Do you have any other changes that need to be made, or I can merge this pull request?
I will rebase my pull request once it's merged. The old travis CI workflow is actually not working, and that's the reason why the duplicate theme name has not been found

TriDiamond commented:
@stevenjoezhang Thanks for remaining me, I double check all the info, found an incorrect Github link for my Aurora theme. It's all good to go now. Thanks for taking your time reviewing this pull request!
